dataland-external-storage/src/main/kotlin/org/dataland/datalandexternalstorage/services/EurodatDataStore.kt

```diff
@@ -3,175 +3,43 @@ package org.dataland.datalandexternalstorage.services
-        }
-        return block()
-    }
-
-    @Suppress("TooGenericExceptionThrown")
-    private fun isSafeDepositBoxAvailable() {
-        if (postSafeDepositBoxCreationRequest().response.contains("Database already exists")) {
-            logger.info("Safe deposit box exists.")
-        } else {
-            throw Exception("Service not there.")
-        }
-    }
-
-    /**
-     * Sends a POST request to the safe deposit box creation endpoint of the EuroDaT client.
-     */
-    fun postSafeDepositBoxCreationRequest(): SafeDepositDatabaseResponse {
-        val creationRequest = SafeDepositDatabaseRequest(eurodatAppName)
-        return safeDepositDatabaseResourceClient.apiV1ClientControllerDatabaseServicePost(creationRequest)
-    }
-
-    /**
-     * Method that listens to the storage_queue and stores data into the database in case there is a message on the
-     * storage_queue
-     * @param payload the content of the message
-     * @param correlationId the correlation ID of the current user process
-     * @param type the type of the message
-     */
-    @RabbitListener(
-        bindings = [
-            QueueBinding(
-                value = Queue(
-                    "requestReceivedEurodatDataStore",
-                    arguments = [
-                        Argument(name = "x-dead-letter-exchange", value = ExchangeName.DeadLetter),
-                        Argument(name = "x-dead-letter-routing-key", value = "deadLetterKey"),
-                        Argument(name = "defaultRequeueRejected", value = "false"),
-                    ],
-                ),
-                exchange = Exchange(ExchangeName.PrivateRequestReceived, declare = "false"),
-                key = [""],
-            ),
-        ],
-    )
-    fun processStorageRequest(
-        @Payload payload: String,
-        @Header(MessageHeaderKey.CorrelationId) correlationId: String,
-        @Header(MessageHeaderKey.Type) type: String,
-    ) {
-        messageUtils.validateMessageType(type, MessageType.PrivateDataReceived)
-        val dataId = JSONObject(payload).getString("dataId")
-        if (dataId.isEmpty()) {
-            throw MessageQueueRejectException("Provided data ID is empty.")
-        }
-        logger.info(
-            "Received storage request for dataId $dataId and correlationId $correlationId with payload: $payload",
-        )
-        messageUtils.rejectMessageOnException {
-            val actionType = JSONObject(payload).getString("actionType")
-            if (actionType == ActionType.StorePrivateDataAndDocuments) {
-                try {
-                    storeDataInEurodat(dataId, correlationId, payload)
-                    sendMessageAfterSuccessfulStorage(payload, correlationId)
-                } catch (ex: SQLException) {
-                    logger.error("A sql exception was thrown: $ex")
-                }
-            }
-        }
-    }
-
     /**
```

Comment:
Dass in die retryWrapperMethod hier immer `getEurodatCredentials` geschrieben wird, ist ja theoretisch gelogen, weil so eine Methode gibt es gar nicht.
Mein Vorschlag ist es, den param `inputMethod` in der Funktion `retryWrapperMethod()` lieber z.B. "operationDescription" oder "descriptionOfOperation" oder sowas zu nennen, und da einfach immer reinzuschreiben, was im retryWrapper passiert.

Also in diesem Fall vllt sowas wie "getting JDBC credentials from EuroDaT" oder sowas.

Reply:
Geändert